components/calendar/calendar.js

```diff
@@ -0,0 +1,474 @@
+				}
+
+				table {
+					border-collapse: separate;
+					border-spacing: 0;
+					table-layout: fixed;
+					width: 100%;
+				}
+
+				th[role="columnheader"] {
+					padding: 0.45rem 0;
+				}
+
+				th > abbr {
+					color: var(--d2l-color-tungsten);
+					display: block;
+					font-size: 0.8rem;
+					font-weight: normal;
+					text-align: center;
+					text-decoration: none;
+				}
+
+				thead {
+					vertical-align: top;
+				}
+
+				.d2l-calendar {
+					border-radius: 4px;
+				}
+
+				.d2l-calendar-title {
+					border-top-left-radius: 4px;
+					border-top-right-radius: 4px;
+					text-align: center;
+				}
+
+				.d2l-calendar-title .d2l-heading-4 {
+					margin: 0.45rem 0 1.25rem 0;
+				}
+			`];
+	}
+
+	static get resources() {
+		return {
+			'ar': { show: 'Show {month} ' },
+			'en': { show: 'Show {month}' },
+			'es': { show: 'Show {month} ' },
+			'fr': { show: 'Show {month} ' },
+			'ja': { show: 'Show {month} ' },
+			'ko': { show: 'Show {month} ' },
+			'nl': { show: 'Show {month} ' },
+			'pt': { show: 'Show {month} ' },
+			'sv': { show: 'Show {month} ' },
+			'tr': { show: 'Show {month} ' },
+			'zh': { show: 'Show {month} ' },
+			'zh-tw': { show: 'Show {month} ' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		const descriptor = getDateTimeDescriptor();
+		this._monthNames = descriptor.calendar.months.long;
+		this._firstDayOfWeek = descriptor.calendar.firstDayOfWeek;
+		const weekdays = descriptor.calendar.days.long;
+		const weekdaysShort = descriptor.calendar.days.short;
+
+		this._arrangedWeekdays = [];
+		for (let i = this._firstDayOfWeek; i < this._firstDayOfWeek + 7; i++) {
+			const index = i % 7;
+			const day = {
+				long: weekdays[index],
+				short: weekdaysShort[index]
+			};
+			this._arrangedWeekdays.push(day);
+		}
+
+		this._dialogLabelId = getUniqueId();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.addEventListener('keydown', this._onKeyDown);
+		this.addEventListener('d2l-calendar-selected', (e) => {
+			this._selectedDate = e.detail.date;
+		});
+
+		let selected;
+		if (this.selectedValue) {
+			selected = parseDate(this.selectedValue);
+		} else {
+			selected = new Date();
+		}
+
+		this._selectedDate = {
+			month: selected.getMonth(),
+			date: selected.getDate(),
+			year: selected.getFullYear()
+		};
+
+		this._shownMonth = this._selectedDate.month;
+		this._shownYear = this._selectedDate.year;
+
+		let focusDay = 1;
+		if (this._selectedDate.month === this._shownMonth) focusDay = this._selectedDate.date;
+		this._focusDate = {
+			date: focusDay,
+			month: this._shownMonth,
+			year: this._shownYear
+		};
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === '_selectedDate') {
+				this._generateDaysInMonth();
+			}
+			if (prop === '_shownMonth') {
+				this._onMonthChange();
+				this._generateDaysInMonth();
+				this.updateComplete.then(() => {
```

Comment:
Yeah this definitely seems odd... especially since there doesn't appear to be any change between these calls.

Aside, it would be valid to make this method `async` (for the `forEach`) and use `await` to avoid the nesting here.